pkg/3scale/amp/operator/upgrade.go

```diff
@@ -521,13 +527,14 @@ func (u *UpgradeApiManager) upgradeAwsSecret() (reconcile.Result, error) {
 func (u *UpgradeApiManager) upgradeApimanagerCRS3Attrs() (reconcile.Result, error) {
 	changed := false
 
-	if u.Cr.Spec.System.FileStorageSpec.S3.AWSBucket != "" {
-		u.Cr.Spec.System.FileStorageSpec.S3.AWSBucket = ""
-		changed = true
-	}
-
-	if u.Cr.Spec.System.FileStorageSpec.S3.AWSRegion != "" {
-		u.Cr.Spec.System.FileStorageSpec.S3.AWSRegion = ""
+	if u.Cr.Spec.System.FileStorageSpec.DeprecatedS3 != nil {
```

Comment:
We now migrate the old section content to the new section

Reply:
see comments above. You can assume it is already not nil. As it should be the last step of the upgrade procedure

Reply:
Yes I was unsure of this, just added it in case something goes wrong in some of the previous steps